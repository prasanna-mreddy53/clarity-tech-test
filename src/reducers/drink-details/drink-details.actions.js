import { drinkDetailsActionTypes } from "./drink-details.types";

export const fetchDetailsRequest = () => ({
  type: drinkDetailsActionTypes.FETCH_DETAILS_REQUEST,
});

export const fetchDetailsSuccess = (json) => ({
  type: drinkDetailsActionTypes.FETCH_DETAILS_SUCCESS,
  payload: json,
});

export const fetchDetailsFailure = (error) => ({
  type: drinkDetailsActionTypes.FETCH_DETAILS_FAILURE,
  payload: error,
});

export const fetchDrinkDetails = (drinkId) => {
  return async (dispatch) => {
    dispatch(fetchDetailsRequest);
    try {
      let response = await fetch(
        `https://www.thecocktaildb.com/api/json/v1/1/lookup.php?i=${drinkId}`
      );
      let json = await response.json();
      dispatch(fetchDetailsSuccess(json));
    } catch (error) {
      dispatch(fetchDetailsFailure(error));
    }
  };
};
