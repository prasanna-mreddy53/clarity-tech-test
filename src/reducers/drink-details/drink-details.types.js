export const drinkDetailsActionTypes = {
  FETCH_DETAILS_REQUEST: "FETCH_DETAILS_REQUEST",
  FETCH_DETAILS_SUCCESS: "FETCH_DETAILS_SUCCESS",
  FETCH_DETAILS_FAILURE: "FETCH_DETAILS_FAILURE",
};
