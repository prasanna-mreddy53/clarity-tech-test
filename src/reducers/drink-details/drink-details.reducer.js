import { drinkDetailsActionTypes } from "./drink-details.types";

const INITIAL_STATE = {
  loading: false,
  drinkDetails: [],
  error: "",
};

const drinkDetailsReducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case drinkDetailsActionTypes.FETCH_DETAILS_REQUEST:
      return {
        ...state,
        loading: true,
      };
    case drinkDetailsActionTypes.FETCH_DETAILS_FAILURE:
      return {
        ...state,
        error: action.payload,
      };
    case drinkDetailsActionTypes.FETCH_DETAILS_SUCCESS:
      return {
        ...state,
        drinkDetails: action.payload,
      };
    default:
      return state;
  }
};

export default drinkDetailsReducer;
