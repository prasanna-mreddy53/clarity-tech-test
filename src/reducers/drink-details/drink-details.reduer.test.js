import reducer from "./drink-details.reducer";
import { drinkDetailsActionTypes } from "./drink-details.types";

describe("Drink Details reducer", () => {
  it("Drink Details should return the initial state", () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it("should return the initial state with default values", () => {
    expect(reducer(undefined, {})).toEqual({
      loading: false,
      drinkDetails: [],
      error: "",
    });
  });

  it("should match FETCH_DETAILS_REQUEST action", () => {
    expect(
      reducer([], {
        type: drinkDetailsActionTypes.FETCH_DETAILS_REQUEST,
      })
    ).toEqual({
      loading: true,
    });
  });

  it("should match FETCH_DETAILS_SUCCESS action", () => {
    expect(
      reducer(
        { drinkDetails: [], loading: false, error: "" },
        {
          type: drinkDetailsActionTypes.FETCH_DETAILS_SUCCESS,
          drinkDetails: { drinks: [{ strDrink: "Margarita" }] },
        }
      )
    ).toEqual({
      drinkDetails: [],
      loading: false,
      error: "",
      drinkDetails: undefined,
    });
  });

  it("Drink Details FETCH_DETAILS_SUCCESS action", () => {
    expect(
      reducer(
        {
          drinkDetails: { drinks: [{ strDrink: "Margarita" }] },
        },
        drinkDetailsActionTypes.FETCH_DETAILS_SUCCESS
      )
    ).toMatchSnapshot();
  });

  it("should match FETCH_DETAILS_FAILURE action", () => {
    expect(
      reducer([{ drinkDetails: [], loading: false, error: "" }], {
        type: drinkDetailsActionTypes.FETCH_DETAILS_FAILURE,
        error: "could not load details",
      })
    ).toMatchSnapshot();
  });
});
