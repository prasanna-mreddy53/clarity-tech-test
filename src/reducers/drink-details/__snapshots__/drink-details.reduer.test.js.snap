// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drink Details reducer Drink Details FETCH_DETAILS_SUCCESS action 1`] = `
Object {
  "drinkDetails": Object {
    "drinks": Array [
      Object {
        "strDrink": "Margarita",
      },
    ],
  },
}
`;

exports[`Drink Details reducer Drink Details should return the initial state 1`] = `
Object {
  "drinkDetails": Array [],
  "error": "",
  "loading": false,
}
`;

exports[`Drink Details reducer should match FETCH_DETAILS_FAILURE action 1`] = `
Object {
  "0": Object {
    "drinkDetails": Array [],
    "error": "",
    "loading": false,
  },
  "error": undefined,
}
`;
